### DISCUSSION TITLE
How to put a widget over another

### DISCUSSION THREAD

**Discussion by snylonue** [hecrj/iced#998] [Q&A] [answered]
For example, I want to show some texts over an image.
I tried `Tooltip` but it didn't satisfy my needs.

**Comment by tarkah** [comment, marked as answer]
The best place to point you for now is to look at how the [PickList](https://github.com/hecrj/iced/blob/master/native/src/widget/pick_list.rs) is implemented. If a [Widget returns an](https://github.com/hecrj/iced/blob/master/native/src/widget.rs#L178-L183) ([Overlay](https://github.com/hecrj/iced/blob/master/native/src/overlay.rs)), that will rendered over everything else.

It's important to note that currently Iced only supports having a single `Overlay` up at one time, and an `Overlay` can't nest over another `Overlay`. However, you can definitely build something very complex if you're just adding normal elements to the `Overlay`.

You can also check out https://github.com/iced-rs/iced_aw. A bunch of additional widgets have been created there, such as a `Modal`, that use `Overlay`.

**Reply by DavidBJaffe** [reply]
1. Could one use a Widget that returns an Overlay to display text on top of "everything else", in such a fashion that what was below the widget is completely hidden?

2. Can that Widget be positioned in the window at given coordinates?

**Reply by 90-008** [reply]
1. The tooltip widget does exactly what you want (display some text over other widgets). Although #749 might be an issue.
2. You could do this if you made your own widget and then you would be able to specify any position you want for the overlay that it will create.

**Comment by DavidBJaffe** [comment]
Yes that's helpful, thank you.  A Tooltip can be positioned in any of several places relative to another widget, whereas I'd like to provide absolute coordinates for where it should go.  So perhaps this would require creation of a new Tooltip-like widget.

I've decided to go with an easier solution, involving configurable positioning of a "tooltip" box within a canvas.
